src/modules/ekf2/ekf2_main.cpp

```diff
@@ -1174,7 +1177,13 @@ void Ekf2::run()
 						}
 					}
 
-					if (range_finder_updated) { _ekf.setRangeData(range_finder.timestamp, range_finder.current_distance); }
+					if (range_finder_updated) {
+						if (range_finder.current_distance <= _gnd_effect_deadzone.get()) {
```

Comment:
@EliaTarasov We should not directly use the range finder data but instead the distance to ground estimate from the estimator. You can just move the activation of the ground effect further down where we publish local position. The code could then look like:
`if (lpos.dist_bottom_valid && lpos.dist_bottom < _gnd_effect_deadzone.get() < 1.0f)`